### PR TITLE
Magiclysm - Magic Items I - Wearables

### PR DESCRIPTION
**Pull request by foulman**
A large collection of new magic items, mostly worn items like rings, bracers, belts, boots, many utilizing the new artifact code.

#### Summary
```SUMMARY: Mods "Magiclysm - New Magic Items I - Wearables"```

#### Purpose of change
Introduces numerous new magic items, and will spread them out to be found in the world in assorted loot tables (this still needs to be worked out).

#### Additional context
This utilizes some of the new functionality @KorGgenT has created for artifacts, such as attribute bonuses, speed, movement cost.

### CODE REVIEW

**Comment by John-Candlebury** [comment]
Awesome work!

File: data/mods/Magiclysm/enchanted/enchanted_rings.json [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    "coverage": 0,
+    "warmth": 0,
+    "flags": [ "WATER_FRIENDLY", "ALLOWS_NATURAL_ATTACKS", "FANCY", "ONLY_ONE", "SKINTIGHT" ]
+  },
+  {
+    "abstract": "mring_gold",
+    "type": "TOOL_ARMOR",
+    "name": "magic ring",
+    "description": "A generic gold magic ring.",
+    "weight": "6 g",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_rings.json]:
```suggestion
    "weight": "9 g",
```

File: data/mods/Magiclysm/enchanted/enchanted_rings.json [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    "coverage": 0,
+    "warmth": 0,
+    "flags": [ "WATER_FRIENDLY", "ALLOWS_NATURAL_ATTACKS", "FANCY", "ONLY_ONE", "SKINTIGHT" ]
+  },
+  {
+    "abstract": "mring_platinum",
+    "type": "TOOL_ARMOR",
+    "name": "magic ring",
+    "description": "A generic platinum magic ring.",
+    "weight": "6 g",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_rings.json]:
```suggestion
    "weight": "11.7 g",
```

File: data/mods/Magiclysm/enchanted/enchanted_rings.json [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+    "coverage": 0,
+    "warmth": 0,
+    "flags": [ "WATER_FRIENDLY", "ALLOWS_NATURAL_ATTACKS", "FANCY", "ONLY_ONE", "SKINTIGHT" ]
+  },
+  {
+    "abstract": "mring_silver",
+    "type": "TOOL_ARMOR",
+    "name": "magic ring",
+    "description": "A generic silver magic ring.",
+    "weight": "6 g",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_rings.json]:
```suggestion
    "weight": "5.4 g",
```

File: data/mods/Magiclysm/enchanted/enchanted_rings.json [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+  },
+  {
+    "id": "conj_throwing_blade1",
+    "type": "SPELL",
+    "name": "Conjure Throwing Blade I",
+    "description": "conjures 3 throwing knives",
+    "valid_targets": [ "self" ],
+    "effect": "spawn_item",
+    "effect_str": "throwing_knife",
+    "message": "You activate your ring and 3 throwing knives appear, ready to throw!",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_rings.json]:
```suggestion
    "message": "You activate your ring and three throwing knives appear, ready to throw!",
```

File: data/mods/Magiclysm/enchanted/enchanted_rings.json [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+    "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "PERCEPTION", "add": 4 } ] } ] }
+  },
+  {
+    "copy-from": "mring_copper",
+    "type": "GENERIC",
+    "id": "mring_speed_3",
+    "name": "ring of speed +3",
+    "name_plural": "rings of speed +3",
+    "description": "A copper ring that makes you a little stronger when you wear it.",
+    "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "SPEED", "add": 3 } ] } ] }
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_rings.json]: Is SPEED both movecost and attackcost?

File: data/mods/Magiclysm/enchanted/enchanted_bracers.json [hunk 6]
```diff
@@ -0,0 +1,119 @@
+[
+  {
+    "id": "armguard_larmor",
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_bracers.json]: Ah! You're right, I commonly cut and paste that kinda thing in there as a reference when I start figuring out what the base should be, my bad. I'll clean those up.

File: data/mods/Magiclysm/enchanted/enchanted_bracers.json [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    "name": "greater bracer of lightning",
+    "name_plural": "greater bracers of lightning",
+    "initial_charges": 5,
+    "charges": 5,
+    "max_charges": 5,
+    "charges_per_use": 1,
+    "artifact_data": { "charge_type": "ARTC_TIME" },
+    "flags": [ "BELTED", "STURDY", "BLOCK_WHILE_WORN" ],
+    "techniques": [ "WBLOCK_1" ],
+    "use_action": { "type": "cast_spell", "spell_id": "lightning_bolt", "no_fail": true, "level": 15, "need_worn": true }
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_bracers.json]: Yes, purely intentional. A more powerful effect and more charges.

File: data/mods/Magiclysm/enchanted/enchanted_masks.json [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  {
+    "type": "TOOL_ARMOR",
+    "id": "mmask_disappearance",
+    "name": "mask of disappearance",
+    "plural": "masks of disappearance",
+    "description": "A mask with no facial features at all, just eye and mouth holes, upon activation it makes the everything ignore your presence for a while.",
+    "copy-from": "mmask",
+    "initial_charges": 10,
+    "max_charges": 10,
+    "charges_per_use": 5,
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_masks.json]: Yep. Only way.

File: data/mods/Magiclysm/enchanted/enchanted_rings.json (hunk 5 above)
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_rings.json]: Yes, speed is a general figure for everything. Right now there's no attack_cost, when there is, I'll create attack_cost specific items.

File: data/mods/Magiclysm/enchanted/enchanted_misc.json [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+    "material": [ "steel" ],
+    "symbol": ";",
+    "color": "red",
+    "ammo": "thread",
+    "sub": "sewing_kit",
+    "initial_charges": 500,
+    "max_charges": 1000,
+    "charges_per_use": 1,
+    "artifact_data": { "charge_type": "ARTC_TIME" },
+    "qualities": [ [ "SEW", 2 ], [ "KNIT", 2 ], [ "CUT", 2 ] ],
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_misc.json]: Yeah, it works okay as is, and if/when that gets done it should be ready for them as the best thing on the block to sew with! I also figured once we start getting more progress in some other things (more artifact stuff), we could get into making your own enchanted clothing, so we might require it ourselves.
You and I should probably put our heads together on that at some point once Korg gets a bit more done.

**Comment by foulman** [comment]
> 
> 
> Awesome work!

Why thank you, good sir! A great compliment from the guy that reworked teleportation!
Did you by any chance read my last comment on there, I'm curious how doable that is or if you would be interested.

File: data/mods/Magiclysm/enchanted/enchanted_tokens.json [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+    "symbol": "o",
+    "color": "light_gray"
+  },
+  {
+    "copy-from": "mtoken",
+    "type": "GENERIC",
+    "id": "mtoken_longsword",
+    "name": "longsword token",
+    "name_plural": "longsword tokens",
+    "description": "A large silver coin that, when activted by uttering the word on the back, turns into the item pictured on the front, in this case a longsword.",
```
Comment (aoae23) [inline data/mods/Magiclysm/enchanted/enchanted_tokens.json]: activated

Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_tokens.json]: D'oh, fixed.

File: data/mods/Magiclysm/enchanted/enchanted_rings.json [hunk 11]
```diff
@@ -0,0 +1,277 @@
...
+    "use_action": { "type": "cast_spell", "spell_id": "conj_throwing_blade1", "no_fail": true, "level": 1, "need_worn": true },
+    "initial_charges": 5,
+    "charges": 5,
+    "max_charges": 5,
+    "charges_per_use": 1,
+    "artifact_data": { "charge_type": "ARTC_TIME" }
+  },
+  {
+    "id": "conj_throwing_blade1",
+    "type": "SPELL",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/enchanted/enchanted_rings.json]: i'd prefer that spells stay in the spell folder. i believe there is already a json file for spells activated from objects, as well.

File: data/mods/Magiclysm/enchanted/enchanted_masks.json [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    "coverage": 95,
+    "encumbrance": 10,
+    "warmth": 10,
+    "material_thickness": 2,
+    "environmental_protection": 1,
+    "flags": [ "WATER_FRIENDLY", "STURDY" ]
+  },
+  {
+    "type": "TOOL_ARMOR",
+    "id": "mmask_disappearance",
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_masks.json]: all the magic items I was tagging with an m in front, but I guess it's not necessary, just might be easier to search for stuff later on down the road when there's a ton of them.

File: data/mods/Magiclysm/enchanted/enchanted_masks.json [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+    "id": "mmask_vision",
+    "name": "mask of perfect vision",
+    "plural": "masks of perfect vision",
+    "description": "A decidedly steampunk-looking half mask that covers the eye area of the face, it has large lenses that correct and greatly enhance the vision of the wearer.",
+    "copy-from": "mmask",
+    "covers": [ "EYES" ],
+    "coverage": 100,
+    "encumbrance": 5,
+    "artifact_data": { "charge_type": "ARTC_TIME" },
+    "flags": [ "FANCY", "WATER_FRIENDLY", "SUN_GLASSES", "FIX_NEARSIGHT", "FIX_FARSIGHT" ],
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_masks.json]: actually, I did want to add in binocular and maybe even infra or nv, cutting perception down would be a fine exchange. these should be a big prize.

File: data/mods/Magiclysm/enchanted/enchanted_rings.json (hunk 11 above)
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_rings.json]: I just kept it there for reference, it'll be removed and split off into a file.

File: data/mods/Magiclysm/enchanted/enchanted_belts.json [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+    "relic_data": {
+      "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "STRENGTH", "multiply": 1.0 } ] } ]
+    }
+  },
+  {
+    "type": "TOOL_ARMOR",
+    "copy-from": "mbelt_leather",
+    "id": "mbelt_pockets",
+    "name": "Girdle of Pockets",
+    "description": "A wide girdle that fits around your waist, coverd in numerous small pouches that hold a lot more than they should.",
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_belts.json]: Oh I will be reworking a LOT of things when you get that done, no worries.

File: data/mods/Magiclysm/enchanted/enchanted_bracers.json [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+    "name": "bracer of lesser defense",
+    "name_plural": "bracers of lesser defense",
+    "material_thickness": 6,
+    "initial_charges": 3,
+    "charges": 3,
+    "max_charges": 3,
+    "charges_per_use": 1,
+    "artifact_data": { "charge_type": "ARTC_TIME" },
+    "flags": [ "BELTED", "STURDY", "BLOCK_WHILE_WORN" ],
+    "techniques": [ "WBLOCK_3" ]
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_bracers.json]: Oh crap, how embarrassing. This must be one of the things I forgot to finish, I probably jumped to another type of items. I'll fix it. Was going to have it activate to give you an aura.

File: data/mods/Magiclysm/enchanted/enchanted_rings.json [hunk 16]
```diff
@@ -267,11 +267,79 @@
...
     "description": "A platinum ring that makes you a little stronger when you wear it.",
     "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "values": [ { "value": "SPEED", "add": 10 } ] } ] }
+  },
+  {
+    "copy-from": "mring_copper",
+    "type": "TOOL_ARMOR",
+    "id": "mring_protection_2",
+    "name": "ring of protection +2",
+    "name_plural": "rings of protection +2",
+    "description": "A copper ring that makes you a little stronger when you wear it.",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_rings.json]: Are you going to give the rings of X different descriptions?

File: data/mods/Magiclysm/enchanted/enchanted_tokens.json [hunk 17]
```diff
@@ -74,16 +74,91 @@
...
+  {
+    "copy-from": "mtoken",
+    "type": "GENERIC",
+    "id": "mtoken_screwdriver",
+    "name": "screwdriver token",
+    "name_plural": "screwdriver tokens",
+    "description": "A large silver coin that, when activated by uttering the word on the back, turns into the item pictured on the front, in this case a screwdriver.",
+    "use_action": {
+      "type": "transform",
+      "target": "screwdriver",
```
Comment (LaVeyanFiend) [inline data/mods/Magiclysm/enchanted/enchanted_tokens.json]: This seems like a useless item due to how easy you can make a screwdriver, perhaps change to screwdriver set?

File: data/mods/Magiclysm/enchanted/enchanted_belts.json [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+  },
+  {
+    "type": "TOOL_ARMOR",
+    "copy-from": "mbelt_leather",
+    "id": "mbelt_weaponry",
+    "name": "Belt of Weaponry",
+    "description": "A wide girdle that fits around your waist, you can sheath or holster any weapon into it in the blink of an eye, and it seemingly stores them somewhere else.",
+    "coverage": 10,
+    "encumbrance": 3,
+    "storage": "10 L",
```
[the remark this answers is not in the document]
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_belts.json]: removed

File: data/mods/Magiclysm/enchanted/enchanted_tokens.json (hunk 17 above)
Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_tokens.json]: Excellent idea, they were thrown in at the last minute, though it is intended for them to be somewhat common items to sprinkle around.

File: data/mods/Magiclysm/enchanted/enchanted_melee.json [hunk 19]
```diff
@@ -136,7 +136,7 @@
   },
   {
     "type": "GENERIC",
-    "id": "metal_bat_plus_two",
+    "id": "bat_metal_plus_two",
```
Comment (kevingranade) [inline data/mods/Magiclysm/enchanted/enchanted_melee.json]: This is going to break saves where the item isn't present.
If it's WIP and regular games can't have it yet anyway that's fine, but if there's an in-game way to acquire it, it's not ok and needs a migration entry.

Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_melee.json]: They never spawned, wasn't in loot lists until now. we have our own obsolete file for things like that.

Comment (foulman) [reply data/mods/Magiclysm/enchanted/enchanted_melee.json]: although... hey @LaVeyanFiend , you might want to check and see if those metal bats were used in your MA thing, you may remember off the top of your head. One was a typo.

Comment (LaVeyanFiend) [reply data/mods/Magiclysm/enchanted/enchanted_melee.json]: Baseball bats aren't used in any MA, so I wouldn't have copied the magical ones in.

**Comment by KorGgenT** [comment]
is this PR part of all the ones that have already been merged now?

**Comment by foulman** [comment]
> 
> 
> is this PR part of all the ones that have already been merged now?

This is purely here for reference at this point, all that is left is reworking the itemgroups for changes that have been made from the previous PRs this was split into. I'll remove it when I get that submitted, hopefully today if I can get my car problem sorted out.

**Pull request closed without merging**